sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DisposeNotImplementingDispose.CSharp9.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+
+void Dispose() { } // top level local function, compliant
+
+public record R
+{
+    void Dispose() { } //Noncompliant {{Either implement 'IDisposable.Dispose', or totally rename this method to prevent confusion.}}
+}
+
+public record GarbageDisposalExceptionBase : IDisposable
+{
+    protected virtual void Dispose(bool disposing)
+    {
+        //...
+    }
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+}
+
+public partial record MyPartial : IDisposable
+{
+    public void Dispose()
+    {
+        // Dispose(10)
+    }
+}
+
+public partial record MyPartial
+{
+    public void Dispose(int i) // FN, partial classes are not processed, see https://github.com/dotnet/roslyn/issues/3748
```

Comment:
I don't understand this case. What I see:
There's a `Dispose()` method implementing `IDisposable` for this class. So it's not a FN.
`Dispose(int i)` has similar name, but probably should not have an effect on the test (I haven't seen standalone overload like this in the old test).

Reply:
this was actually a copy paste of the test in `TestCases\DisposeNotImplementingDispose.cs` ... indeed